### PR TITLE
Fix for: Paste buttons has wrong state in read-only mode

### LINKED ISSUE #2775: divarea editor, when setReadOnly is triggered by mouseclick event paste buttons has wrong state
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open sample: https://codepen.io/anon/pen/RvwJLL
2. Check 'Make editor read only'.

### Expected result

Paste buttons are disabled.

### Actual result

Paste buttons are not disabled

This happens because `mouseup` listener on editor.document doesn't take in count editor being read only:

https://github.com/ckeditor/ckeditor-dev/blob/d931efee7bb8875e489c172fa3e20de8e8209aef/plugins/clipboard/plugin.js#L742

https://github.com/ckeditor/ckeditor-dev/blob/d931efee7bb8875e489c172fa3e20de8e8209aef/plugins/clipboard/plugin.js#L1212-L1222

## Other details

This might affect inline editor, however focusing editor makes editor update buttons state, hence it is not visible when toolbar is hidden.

* Installed CKEditor plugins: 'divarea'

### PR DESCRIPTION
**Pull request by engineering-this**
## What is the purpose of this pull request?

Bugfix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [X] Unit tests
- [X] Manual tests

## What changes did you make?

Included `editor.readOnly` in check for command state function.

Closes #2775

### CODE REVIEW

File: plugins/clipboard/plugin.js [hunk 1]
```diff
@@ -1222,11 +1222,14 @@
 		}
 
 		function stateFromNamedCommand( command ) {
-			if ( inReadOnly && command in { paste: 1, cut: 1 } )
+			// We need to correctly update toolbar states on readOnly (#2775).
+			if ( ( inReadOnly || editor.readOnly ) && command in { paste: 1, cut: 1 } ) {
```
Comment (Comandeer) [inline plugins/clipboard/plugin.js]: Won't it be better to ensure that `inReadOnly` variable contains correct info?

File: tests/plugins/clipboard/readonly.js [hunk 2]
```diff
@@ -58,4 +58,47 @@ var tests = {
 
 tests = bender.tools.createTestsForEditors( CKEDITOR.tools.objectKeys( bender.editors ), tests );
 
+tests = CKEDITOR.tools.object.merge( tests, {
```
Comment (Comandeer) [inline tests/plugins/clipboard/readonly.js]: You can simply:

```suggestion
tests[ 'test paste command state in divarea editor' ] = function() {
```

File: tests/plugins/clipboard/readonly.js [hunk 3]
```diff
@@ -58,4 +58,47 @@ var tests = {
...
+
+			editor.setReadOnly( true );
+
+			// mouseup listener updates state in timeout, so we need to make test asynchronous.
+			setTimeout( function() {
+				resume( function() {
+					assertCommands( CKEDITOR.TRISTATE_DISABLED, 'true' );
+
+					editor.setReadOnly( false );
+					assertCommands( CKEDITOR.TRISTATE_OFF, 'false.' );
```
Comment (Comandeer) [inline tests/plugins/clipboard/readonly.js]: Such message is totally unhelpful. Please write more descriptive messages.

File: plugins/clipboard/plugin.js (hunk 1 above)
Comment (engineering-this) [reply plugins/clipboard/plugin.js]: This variable is used only here, but it is updated in other places, I'll move it here.

**Comment by f1ames** [comment]
Overtaking this PR review.

**Comment by f1ames** [comment]
Rebased on latest `master`.

**Pull request merged**